rust/core-lib/src/index_set.rs

```diff
@@ -18,23 +18,36 @@
 // Note: this data structure has nontrivial overlap with Subset in the rope
 // crate. Maybe we don't need both.
 
-use std::cmp::{max, min};
+use std::cmp::{max, min, Ordering};
 use xi_rope::{RopeDelta, Transformer};
 
 pub struct IndexSet {
     ranges: Vec<(usize, usize)>,
 }
 
 pub fn remove_n_at<T: Clone>(v: &mut Vec<T>, index: usize, n: usize) {
-    if n == 1 {
-        v.remove(index);
-    } else if n > 1 {
-        let new_len = v.len() - n;
-        for i in index..new_len {
-            v[i] = v[i + n].clone();
+    match n.cmp(&1) {
+        Ordering::Equal => {
+            v.remove(index);
```

Comment:
style nit: I would leave out the braces, and just write this as,

```rust
Ordering::Equal => v.remove(index),
```

Reply:
Get an error `error[E0308]: match arms have incompatible types`
Honestly, I keep it for the time being.

Reply:
oops yep my mistake!